src/lib/core/partialcubeop.test.ts

```diff
@@ -0,0 +1,197 @@
+import * as partial from "./partialcubeop";
+
+const CUBE_FACE_3X3 = {
+  dimension: 3,
+  topLeft: {
+    face: 0,
+    row: 0,
+    column: 0,
+  },
+  top: {
+    face: 0,
+    row: 0,
+    column: 1,
+  },
+  topRight: {
+    face: 0,
+    row: 0,
+    column: 2,
+  },
+  left: {
+    face: 0,
+    row: 1,
+    column: 0,
+  },
+  center: {
+    face: 0,
+    row: 1,
+    column: 1,
+  },
+  right: {
+    face: 0,
+    row: 1,
+    column: 2,
+  },
+  bottomLeft: {
+    face: 0,
+    row: 2,
+    column: 0,
+  },
+  bottom: {
+    face: 0,
+    row: 2,
+    column: 1,
+  },
+  bottomRight: {
+    face: 0,
+    row: 2,
+    column: 2,
+  },
+};
+
+test("3x3 clockwise rotations", () => {
```

Comment:
more of a nit on test style, but I would use the test suite functions in Jest and split it up that way. 
e.g.
```js
describe("3x3 rotations", () => {
	it("should rotate clockwise", () => {
		expect(...).toEqual(...);
		...
	});

	it("should rotate counterclockwise", () => {
		expect(...).toEqual(...);
		...
	});
});

describe("2x2 rotations", () => {
	it("should rotate clockwise", () => {
		expect(...).toEqual(...);
		...
	});

	it("should rotate counterclockwise", () => {
		expect(...).toEqual(...);
		...
	});
});
```
`test()` and `it()` I believe are aliased together, so it doesn't matter to much there, but I feel like it helps being a bit more descriptive.

jest docs: https://jestjs.io/docs/en/api#describename-fn

Reply:
Whoa that's neat. Done.